lib/Dialect/StaticLogic/StaticLogicOps.cpp

```diff
@@ -25,3 +29,28 @@ void StaticLogicDialect::initialize() {
 #include "circt/Dialect/StaticLogic/StaticLogic.cpp.inc"
       >();
 }
+
+LogicalResult PipelineOp::constructSchedulingProblem(sched::SchedulerBase &scheduler){
```

Comment:
I've been wondering how this would work. In general it seems we are asking if we can implement an interface for an operation we didn't define (both for this case, as well as for the target operations to declare their properties). I wonder if there is a general MLIR solution to this issue. If possible, it would be great to avoid defining `mydialect.for`, which is basically `affine.for` + this interface.

Reply:
Right, that's kind of a dealbreaker for the OpInterface idea. We could instead provide a "scheduling problem builder" for relevant MLIR constructs, e.g. for an `mlir::Block`, we'd take its operations and use the SSA graph to construct the deps, and for an `affine.for`, we could additionally consider the affine analysis.